sympy/physics/vector/point.py

```diff
@@ -471,6 +472,44 @@ def v2pt_theory(self, otherpoint, outframe, fixedframe):
         self.set_vel(outframe, v + (omega ^ dist))
         return self.vel(outframe)
 
+    def _calc_vel(self, frame, pos_vect):
+        # Helper Function for velocty
+        if len(self._pos_dict) == 1:
+        #If len of post_dict is one it implies it only has that point from which it's being called
+        #Stops infinite recursion
+            return False
+        #Copies vel_dict of current point(the one through which it's called, self for this function)
+        self._vel_dict_pos = self._vel_dict.copy()
+        #Loop 1 for checking all points at same level in self's pos_dict(to find nearest point in Tree)
+        for p, p_pos in self._pos_dict.items():
+            try:
+                p_vel = p._vel_dict[frame] #Checks if the point's velocity is defined wrt req frame, if not throws Keyerror
+                p_pos.dt(frame) #Checks if pos vector in defined with req frame, if not throws ValueError
+            except KeyError:
+                continue
+            except ValueError:
+                continue
+            self._vel_dict_pos[frame] = p_vel + p_pos.dt(frame) #Updates current point's copied dict
+            return True #Returns True, this point has it's velocity defined in appropriate ref frame
+        for p, p_pos in self._pos_dict.items():
+            #Loop 2 for searching in next level, if not found in same level
```

Comment:
This isn't clearly if this works for a tree depth of 100 or 1000. How can two loops do the breath first search of the entire tree?